### PR TITLE
Feedback form and mobile view UI changes

### PR DESCRIPTION
**Pull request by TheRBajaj**
This PR adds multiple functionalities:

- This PR adds the functionality to enable nl-search through a url parameter so it can be accessed by clicking th `What's New` banner in the web-marketing repo (PR to follow).
- This PR modifies and refines the Feedback form UI feature in order to be ready for release. Feedback form can be accessed from the sidebar or from the mobile view menu.
- This PR also adds mobile view functionality for nl search

# Screen Shot
image
image

# Mobile view Screenshots
image
image
image

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by marinrelatic — APPROVED** [review]

File: src/services/useFeedbackService.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ const useFeedbackService = () => {
 						email: email
 					})
 				});
-				return await response.json();
+				return response;
```
Comment (marinrelatic) [inline src/services/useFeedbackService.ts:28]: How is this response used?
You would want to handle errors if the response.status is not 200.

File: src/hooks/useFeedback.ts [hunk 2]
```diff
@@ -20,9 +20,13 @@ const useFeedback = () => {
 			submitFeedbackForm: async (category: string, rating: number, text: string, email?: string) => {
 				try {
 					setState({ isSubmitting: true });
-					await feedbackService.submitFeedback(category, rating, text, email);
+					const data = await feedbackService.submitFeedback(category, rating, text, email);
+					if (!data.ok) {
```
Comment (marinrelatic) [inline src/hooks/useFeedback.ts:24]: How does this work? Is `ok` a response field?

**Review by marinrelatic — CHANGES_REQUESTED** [review]

File: src/services/useFeedbackService.ts (hunk 1 above)
Comment (TheRBajaj) [reply src/services/useFeedbackService.ts:28]: basically I was encountering a weird bug where 400 or 500 status codes weren't being caught by the try catch statement in `useFeedback.ts`. So to overcome that I return the response and then in `useFeedback.ts` throw an error if we get 400 or 500 status.

File: src/hooks/useFeedback.ts (hunk 2 above)
Comment (TheRBajaj) [reply src/hooks/useFeedback.ts:24]: promise.ok returns a boolean value if 200 was returned. If false, then I throw an error

File: src/atomicui/molecules/FeedbackModal/FeedbackModal.tsx [hunk 3]
```diff
@@ -128,46 +122,65 @@ const FeedbackModal: React.FC<FeedbackModalProps> = ({ open, onClose }) => {
 	const handleSubmit = async () => {
 		const { category, rating, text, email } = formValues;
 		setIsSubmitting(true);
-		await submitFeedbackForm(category, rating, text, email);
+		const response = await submitFeedbackForm(category, rating, text, email);
+		if (response) {
+			clearForm();
+			setShowAlert(true);
+		}
```
Comment (thehoneymad) [inline src/atomicui/molecules/FeedbackModal/FeedbackModal.tsx:129]: This is the success case, right?

File: src/atomicui/molecules/FeedbackModal/FeedbackModal.tsx [hunk 4]
```diff
@@ -128,46 +122,65 @@ const FeedbackModal: React.FC<FeedbackModalProps> = ({ open, onClose }) => {
...
 							{showAlert ? (
-								<Alert variation="success" isDismissible={true} onDismiss={() => dismissAlert()}>
-									Feedback Submitted
+								<Alert
+									variation="success"
+									isDismissible={true}
+									onDismiss={() => dismissAlert()}
+									style={{ borderRadius: "8px" }}
+								>
+									{t("fm__submit_feedback_alert.text")}
```
Comment (thehoneymad) [inline src/atomicui/molecules/FeedbackModal/FeedbackModal.tsx:158]: This is for submit feedback errors.

File: src/atomicui/molecules/FeedbackModal/FeedbackModal.tsx [hunk 5]
```diff
@@ -180,31 +193,54 @@ const FeedbackModal: React.FC<FeedbackModalProps> = ({ open, onClose }) => {
...
+											onClick={() => handleStarClick(star)}
+											style={{
+												cursor: "pointer"
+											}}
+										>
+											{star <= formValues.rating ? <IconStarFilled /> : <IconStar />}
+										</Flex>
+									);
+								})}
+							</Flex>
```
Comment (thehoneymad) [inline src/atomicui/molecules/FeedbackModal/FeedbackModal.tsx:219]: This is the star rating part. We should make a component out of this later.

File: src/atomicui/organisms/SearchBox/SearchBox.tsx [hunk 6]
```diff
@@ -485,21 +490,110 @@ const SearchBox: React.FC<SearchBoxProps> = ({
...
+												marginLeft: "10px"
+											}}
+										/>
+										<Badge size="small" variation="warning" style={{ backgroundColor: "#FF9900", color: "white" }}>
+											{t("prototype.text") as string}
+										</Badge>
+									</Flex>
+								) : (
+									<></>
+								)}
```
Comment (thehoneymad) [inline src/atomicui/organisms/SearchBox/SearchBox.tsx:511]: These are NLP search for mobile.

**Review by thehoneymad — APPROVED** [review]

**Review by marinrelatic — APPROVED** [review]

**Pull request merged**